openmc/deplete/independent_operator.py

```diff
@@ -37,14 +39,20 @@ class IndependentOperator(OpenMCOperator):
 
     .. versionadded:: 0.13.1
 
+    .. versionchanged:: 0.13.4
+        Arguments updated to include list of fluxes and microscopic cross
+        sections.
+
     Parameters
     ----------
     materials : openmc.Materials
         Materials to deplete.
-    micro_xs : MicroXS
-        One-group microscopic cross sections in [b]. If the
-        :class:`~openmc.deplete.MicroXS` object is empty, a decay-only calculation will
-        be run.
+    fluxes : list of numpy.ndarray
```

Comment:
I think if `IndependentOperator` is going to still subclass `OpenMCOperator` then rather than `materials` being the first arg it should be `model` to parallel the `CoupledOperator` class. Then `fluxes` and  `micros` can be optional args that if not specified get computed by `get_microxs_and_flux`, then the optional kwarg `domains` could be added here with a sensible default that defaults to all cells or all materials in the model. What do you think?

Following on to this, I see there is a TODO in `model.py` to support `IndependentOperator`. Could we add that in here with a new kwarg to `Model.deplete` like `operator='coupled'` or `operator='independent'`? Then we could call `model.deplete(timesteps, operator='independent')` and get a cell-based or material-based depletion calculation that runs a single transport calculation if `micros` and `fluxes` aren't already computed somewhere else.

Reply:
I see your point about `OpenMCOperator` but the `IndependentOperator` class is really intended to be something that you can use "independent" of the OpenMC transport solver. `Model` is really a bit too specific to OpenMC as it contains geometry and other stuff that is not really relevant for `IndependentOperator`. That being said, I do like your idea about streamlining things with respect to having fluxes/micros auto-calculated when the user doesn't pass them. Again, I'd prefer leaving that for another PR just to not have too much scope creep here; we can create an issue for this after.

Also agreed that adding support for `IndependentOperator` in the `Model` class could be quite natural with these changes.